languages/python/polar/query.py

```diff
@@ -36,9 +36,10 @@ def run(self):
         """Run the event loop and yield results."""
         assert self.ffi_query, "no query to run"
         while True:
-            event_str = self.ffi_query.next_event().get()
-            event = json.loads(event_str)
+            ffi_event = self.ffi_query.next_event()
+            event = json.loads(ffi_event.get())
             if event == "Done":
```

Comment:
I think this should happen on line 41 after the event is parsed. if it’s here it’ll only be called for done events